### PR TITLE
Rita polish

### PR DESCRIPTION
**Pull request by kingoflolz**
- add api for stats server configuration
- add country code to exit_db
- rename apis to snake case
- clean up error handling in network endpoints
- add general settings api
- add own info api

### CODE REVIEW

File: exit_db/migrations/2018-02-27-043631_add_tables/up.sql [hunk 1]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE clients (
   luci_pass VARCHAR NOT NULL,
   internal_ip VARCHAR NOT NULL,
   email VARCHAR NOT NULL,
-  zip VARCHAR NOT NULL
+  zip VARCHAR NOT NULL,
+  country VARCHAR NOT NULL
```
Comment (drozdziak1) [inline exit_db/migrations/2018-02-27-043631_add_tables/up.sql]: Not all countries have ZIPs, I wonder if we could just convert stuff to coordinates somehow, or a system like what3words, Makaney or Open Location Code

Comment (drozdziak1) [reply exit_db/migrations/2018-02-27-043631_add_tables/up.sql]: Geohash and Open Location Code have Rust implementations

File: rita/src/rita_common/dashboard/mod.rs [hunk 2]
```diff
@@ -158,3 +162,40 @@ impl Handler<GetNodeInfo> for Dashboard {
...
+}
+
+impl Handler<GetStatsServerInfo> for Dashboard {
+    type Result = Result<Option<StatsServerSettings>, Error>;
+
+    fn handle(&mut self, _msg: GetStatsServerInfo, _ctx: &mut Self::Context) -> Self::Result {
+        if SETTING.stats_server_settings_is_set() {
+            Ok(Some(SETTING.get_stats_server_settings().clone()))
+        } else {
+            Ok(None)
```
Comment (drozdziak1) [inline rita/src/rita_common/dashboard/mod.rs]: Can't we just return an `Option` here? This forces the caller to double-unwrap

File: rita/src/rita_common/dashboard/network_endpoints.rs [hunk 3]
```diff
@@ -49,3 +51,30 @@ pub fn get_node_info(req: HttpRequest) -> Box<Future<Item = Json<Vec<NodeInfo>>,
...
+
+pub fn get_stats_server_info(
+    req: HttpRequest,
+) -> Box<Future<Item = Json<Option<StatsServerSettings>>, Error = Error>> {
+    req.body()
+        .from_err()
+        .and_then(move |_: Bytes| {
+            Dashboard::from_registry()
+                .send(GetStatsServerInfo {})
+                .then(move |reply| Ok(Json(reply.unwrap().unwrap())))
```
Comment (drozdziak1) [inline rita/src/rita_common/dashboard/network_endpoints.rs]: Can these get handled?

File: rita/src/rita_common/dashboard/network_endpoints.rs [hunk 4]
```diff
@@ -49,3 +51,30 @@ pub fn get_node_info(req: HttpRequest) -> Box<Future<Item = Json<Vec<NodeInfo>>,
...
+        })
+        .responder()
+}
+
+pub fn set_stats_server_info(req: HttpRequest) -> Box<Future<Item = Json<()>, Error = Error>> {
+    req.body()
+        .from_err()
+        .and_then(move |bytes: Bytes| {
+            let new_settings: Option<StatsServerSettings> =
+                serde_json::from_slice(&bytes[..]).unwrap();
```
Comment (drozdziak1) [inline rita/src/rita_common/dashboard/network_endpoints.rs]: can't this be handled?

File: rita/src/rita_common/dashboard/network_endpoints.rs [hunk 5]
```diff
@@ -49,3 +51,30 @@ pub fn get_node_info(req: HttpRequest) -> Box<Future<Item = Json<Vec<NodeInfo>>,
...
+
+            Dashboard::from_registry()
+                .send(SetStatsServerInfo(new_settings))
+                .then(move |reply| Ok(Json(reply.unwrap().unwrap())))
```
Comment (drozdziak1) [inline rita/src/rita_common/dashboard/network_endpoints.rs]: Same here

File: exit_db/migrations/2018-02-27-043631_add_tables/up.sql (hunk 1 above)
Comment (jkilpatr) [reply exit_db/migrations/2018-02-27-043631_add_tables/up.sql]: I just wanted to call it postal code. As that's pretty relevant. Although some countries don't even have that at which point we can leave it blank. We don't really need postal codes

**Pull request merged**